src/plugins/intel_npu/tools/common/include/tools_helpers.hpp

```diff
@@ -74,74 +74,95 @@ std::map<std::string, std::vector<std::string>> parseInputParameters(std::string
     return return_value;
 }
 
-void boundDynamicShape(std::shared_ptr<ov::Model>& model) {
+/**
+ * @brief Checks the model for dynamism and ensures it is compatible with NPU.
+ *
+ * This function performs several checks on the model's parameters to ensure that
+ * the model's shape and batch size are static and supported by NPU. If the shape
+ * or batch size is dynamic, it throws an exception when the user provides the shape
+ * or batch. Else, it forces the batch size to 1.
+ *
+ * @param model A shared pointer to the OpenVINO model to be checked.
+ * @param shapeOrBatchGiven A boolean flag indicating whether the shape or batch size
+ *                          is provided by the user. Default is true.
+ *
+ * @throws std::logic_error If the model's rank or shape is dynamic and not supported by NPU.
+ */
+void modelDynamismCheck(std::shared_ptr<ov::Model>& model, bool shapeOrBatchGiven = true) {
```

Comment:
I checked the code and I didn't find any use case when we are using `shapeOrBatchGiven =true`. do we really need that parameter?

Reply:
I may need to double check the flow compared to the original to determine if we still need this.